File: _includes/sort-benchmark/algorithms/shell.rs
fn shell_sort(a: &mut [usize]) {
    let mut gap = a.len() / 2;
    while gap > 0 {
        for i in gap..a.len() {
            let x = a[i];
            let mut j = i;
            while j >= gap && a[j - gap] > x {
                a[j] = a[j - gap];
                j -= gap;
            }
            a[j] = x;
        }
        gap /= 2;
    }
}

File: _includes/sort-benchmark/algorithms/shell.swift
func shell_sort(_ a: inout [Int]) {
    a.withUnsafeMutableBufferPointer { shell_sort($0) }
}

func shell_sort(_ a: UnsafeMutableBufferPointer<Int>) {
    var gap = a.count / 2
    while gap > 0 {
        for i in gap..<a.count {
            let x = a[i]
            var j = i
            while j >= gap && a[j - gap] > x {
                a[j] = a[j - gap]
                j -= gap
            }
            a[j] = x
        }
        gap /= 2
    }
}
